### DISCUSSION TITLE
How to access path from Docker image?

### DISCUSSION THREAD

**Discussion by mischakolbe** [community/community#25933] [Actions]
<p>Ciao,<br>
after running a lot of tests I’m hoping someone on here has a hint for me.</p>
<p>I am using a docker image for the 3D software Autodesk Maya:<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/mottosso/docker-maya" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/24309085/181094181-5561d28c-3af3-45b8-9d9b-f0169c505426.png" class="thumbnail onebox-avatar" width="357" height="357">

<h3><a href="https://github.com/mottosso/docker-maya" target="_blank" rel="nofollow noopener">mottosso/docker-maya</a></h3>

<p>Dockerfile for Maya. Contribute to mottosso/docker-maya development by creating an account on GitHub.</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Eventually I need to access Maya’s Python. On Linux machines that’s usually: <code>/usr/autodesk/maya/bin/mayapy</code></p>
<p>My best guess so far is:</p>
<pre><code class="lang-auto">jobs:
  test_maya:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
      
    - name: Load Docker Image
      uses: docker://mottosso/maya:2018
      with:
        path: usr/autodesk/maya
</code></pre>
<p>But for the love of god I can’t access/find where that image was installed, or how to access <code>mayapy</code>. Can someone help me?</p>
<p>Hopefully I didn’t just make a fool of myself :slight_smile: Started using GitHub’s Actions a couple days ago.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/mischakolbe">@mischakolbe</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>You can use <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepswithargs" rel="nofollow noopener"> <strong><code>jobs.&lt;job_id&gt;.steps.with.args</code></strong> </a> to pass the command to the docker image.</p>
<p>To access /user/autodesk/maya folder, please use below format:</p>
<pre><code class="lang-auto">    - name: Load Docker Image
      uses: docker://mottosso/maya:2018
      with:
        args: /bin/sh -c "cd /usr/autodesk/maya &amp;&amp; ls -la"    # access folder and list the files
</code></pre>
<p>My <a href="https://github.com/weide-zhou/ticket13/runs/881262226?check_suite_focus=true#step:5:11" rel="nofollow noopener">workflow</a> for your reference.</p>
<p>I notice the mayapy is set as the default python, you can simply pass the command to the docker image.</p>
<pre><code class="lang-auto">    - name: Load Docker Image
      uses: docker://mottosso/maya:2018
      with:
        args: python -V
</code></pre>
<p>Thanks.</p>

**Comment by mischakolbe** [comment]
<p>You absolute legend, <a class="mention" href="/weide-zhou">@weide-zhou</a>! Thank you so, so much!!</p>
<p>With a couple extra tests this setup is now more streamlined than ever before. If someone told me 1 week ago that this could be all that I need for CI I would have thought they’re nuts:</p>
<pre><code class="lang-auto">name: NoCa CI

# Run tests on any push or pullRequest event
on: [push, pull_request]

jobs:
  test_maya:
    runs-on: ubuntu-latest

    steps:
    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
    - uses: actions/checkout@v2
      
    - name: Load Docker Image
      uses: docker://mottosso/maya:2018
      with:
        args: /usr/autodesk/maya/bin/mayapy -m unittest discover -s ./tests -v
</code></pre>
<p>I noticed too, that <code>python</code> is supposed to be an alias for <code>mayapy</code>. I just did a test and that doesn’t seem to be the case though. But accessing the executable directly works great.</p>
<p>Out of curiosity:<br>
Is the <code>uses</code> kind of its own scope? So any command ran outside of it doesn’t have access to - for example - <code>mayapy</code> anymore?</p>

**Comment by mischakolbe** [comment]
<p>For the very unlikely case someone else needs to test Maya code; this is my latest version.</p>
<p>It’s a bit shabby, because I couldn’t get the <code>uses: docker://mottosso/maya:${{ matrix.version }}</code> to work…<br>
I’ll revisit that again, but at least it works:</p>
<pre><code class="lang-auto">name: NoCa CI

# Run tests on any push or pullRequest event
on: [push, pull_request]

jobs:
  maya2017:
    runs-on: ubuntu-latest
    
    steps:
    - uses: actions/checkout@v2
    - name: Load Docker Image
      uses: docker://mottosso/maya:2017
      with:
        args: /usr/autodesk/maya/bin/mayapy -m unittest discover -s ./tests -v
         
  maya2018:
    runs-on: ubuntu-latest
    
    steps:
    - uses: actions/checkout@v2
    - name: Load Docker Image
      uses: docker://mottosso/maya:2018
      with:
        args: /usr/autodesk/maya/bin/mayapy -m unittest discover -s ./tests -v
         
  maya2019:
    runs-on: ubuntu-latest
    
    steps:
    - uses: actions/checkout@v2
    - name: Load Docker Image
      uses: docker://mottosso/maya:2019
      with:
        args: /usr/autodesk/maya/bin/mayapy -m unittest discover -s ./tests -v
         
  maya2020:
    runs-on: ubuntu-latest
    
    steps:
    - uses: actions/checkout@v2
    - name: Load Docker Image
      uses: docker://mottosso/maya:2020
      with:
        args: /usr/autodesk/maya/bin/mayapy -m unittest discover -s ./tests -v
</code></pre>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/mischakolbe">@mischakolbe</a>,</p>
<p>Thanks for your reply and Glad to know it works for you!</p>
<p><code>uses</code> will pull the actions content or image to your runner, in your case, the command run outside of it doesn’t have access to the mayapy command, since it’s only executed in the docker not hosted runner,  and after the step completed, the docker has been terminated.</p>
<p>variables are not accepted in the <code>uses:image:tag</code> , you have to use hard code. :slightly_smiling_face:</p>
<p>You can mark your answer as a solution since it’s resolved now.</p>
<p>Thanks</p>

**Comment by mischakolbe** [comment]
<p>I marked my last response as the solution.</p>
<p>I read a few things about variables in <code>uses</code>. I can’t fully wrap my head around the reason why that doesn’t work, but I understand that it’s just not possible :slight_smile:</p>
<p>Thank you so much for all your help, <a class="mention" href="/weide-zhou">@weide-zhou</a>!</p>

**Comment by mischakolbe** [comment, marked as answer]
<p>For anyone interested; I wrote a blog post about the pretty straight forward setup I managed to put in place:<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/24309085/181094189-29e255d8-af3e-4135-9bf6-60355cf234dc.png" class="site-icon" width="32" height="32">
      <a href="https://mykolbe.wordpress.com/2020/07/26/continuous-integration-for-maya-scripts/" target="_blank" rel="nofollow noopener" title="06:03AM - 26 July 2020">Mischa Kolbe – 26 Jul 20</a>
  </header>
  <article class="onebox-body">
    <img src="https://mykolbe.files.wordpress.com/2020/07/titleimage_v002.png?fit=440%2C330" class="thumbnail onebox-avatar" width="330" height="330">

<h3><a href="https://mykolbe.wordpress.com/2020/07/26/continuous-integration-for-maya-scripts/" target="_blank" rel="nofollow noopener">Continuous Integration for Maya Scripts</a></h3>

<p>Learn how to automatically validate your Maya Python scripts with GitHub Actions and unittests.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
